test/python/qobj/cpp_measure_opt_flag.json

```diff
@@ -0,0 +1,157 @@
+{
+    "id": "test_measure_opt_flag",
+    "config": {
+        "shots": 5,
+        "seed": 1,
+        "backend_name": "local_qiskit_simulator"
```

Comment:
should be `"backend_name": "local_qasm_simulator_cpp"`

Reply:
I think this file was included by mistake. It doesn't exist on master now, and I probably grabbed it from an old version of your branch. can you please see if it needs to stay, or be deleted? @chriseclectic

Reply:
It needs to stay. I added it in my commit to this PR. It's used to test that sampling multiple measurement outcomes from a single shot happens when it should, and that full simulation of each shot happens in the cases where this optimisation isn't possible.